iree/compiler/Dialect/VMLA/Conversion/VMLAToVM/test/conversion.mlir

```diff
@@ -50,7 +50,7 @@ func @shapeExpansion(%arg0 : !vmla.buffer, %arg1 : index, %arg2 : !vmla.buffer,
   // CHECK-DAG: %c4 = vm.const.i32 4 : i32
   // CHECK-DAG: %c8 = vm.const.i32 8 : i32
   // CHECK-NEXT: vm.call.variadic @vmla.transpose.x16(%arg0, [%c4, %arg1, %c8], [%c1], %arg2, [%arg3, %c4, %c8]) : (!vm.ref<!vmla.buffer>, i32..., i32..., !vm.ref<!vmla.buffer>, i32...)
-  "vmla.transpose"(%arg0, %rs0, %arg2, %rs1) { permutation = dense<[1]> : tensor<1xi32>, element_type = i16 } : (!vmla.buffer, !shapex.ranked_shape<[4,?,8]>, !vmla.buffer, !shapex.ranked_shape<[?,4,8]>) -> ()
+  vmla.transpose %arg0, %rs0, %arg2, %rs1 { permutation = dense<[1]> : tensor<1xi32>} : (!shapex.ranked_shape<[4,?,8]>, !shapex.ranked_shape<[?,4,8]>) i16
```

Comment:
While looking at these examples one thing that springs to mind is that it'd be nice to be able to tell which of the arguments were output parameters. Here it's %arg2, but that's not clear from the IR alone and instead just a convention. C# uses the `out` keyword and that should be possible to support here with the assemblyFormat, so:

```mlir
vmla.transpose %arg0(%rs0 : shapex.ranked_shape...), out %arg2(%rs1 : shapex.ranked_shape...)
```

This way we can quickly scan the IR and see what buffers are being modified vs. ones that are read-only.
(can be a follow-on if that's easier)

Reply:
So something like this?
```mlir
vmla.clamp %arg0, %arg1, %arg2, output %arg3 : f32
```
This would be nice to have across the dialect. I think that an `output` keyword would read a bit more fluidly, and matches `return` not being abbreviated to `ret` in MLIR.

Reply:
let's stick with out as it's the common one used in languages and style